src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -364,21 +364,21 @@ public function slice($offset = 0, $length = -1)
     }
 
     /**
-     * Reduces the list of nodes by calling an anonymous function.
+     * Reduces the list of nodes by calling a callback.
      *
-     * To remove a node from the list, the anonymous function must return false.
+     * To remove a node from the list, the callback must return false.
      *
-     * @param \Closure $closure An anonymous function
+     * @param callable $fn A callback
      *
      * @return Crawler A Crawler instance with the selected nodes.
      *
      * @api
      */
-    public function reduce(\Closure $closure)
+    public function reduce(callable $fn)
     {
         $nodes = array();
         foreach ($this as $i => $node) {
-            if (false !== $closure(new static($node, $this->uri, $this->baseHref), $i)) {
+            if (false !== $fn(new static($node, $this->uri, $this->baseHref), $i)) {
```

Comment:
this does not work for all types of callables. There is still 1 type of callable passing the typehint but failing for the direct call